@here/harp-transfer-manager/src/DeferredPromise.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ *
+ * DeferredPromise takes an executor function for executing it later, when [[exec]] is called.
+ * This class allows wrapping other promises or long running functions for later execution.
+ * @internal
+ */
+export class DeferredPromise<T> {
```

Comment:
I refactored some of the the `TransferManager` code because I initially thought that the bug is somewhere in this class. Some of the changes increase test coverage of this part of the code or make it easier to debug. But I can move it to another PR.

Reply:
Actually, a separate commit is good enough, because you can merge the PR by rebasing, as opposed to merging, but if you do that, you will need to merge parts of the current two commits, because one contains part that you don't want to have in the mainline (the canEvict method for example).

Reply:
I usally click on "squash and merge". This would also get rid of the uncessary `canEvict`change.